matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/RoomDecryptorProvider.kt

```diff
@@ -74,7 +74,7 @@ internal class RoomDecryptorProvider @Inject constructor(
                     this.newSessionListener = object : NewSessionListener {
                         override fun onNewSession(roomId: String?, senderKey: String, sessionId: String) {
                             // PR reviewer: the parameter has been renamed so is now in conflict with the parameter of getOrCreateRoomDecryptor
-                            newSessionListeners.forEach {
+                            newSessionListeners.toList().forEach {
```

Comment:
this could create a lot of unnecessary lists however it seems better than causing `addNewSessionListener/removeSessionListener` to ANR by synchronising them

Reply:
I think the List `newSessionListeners` (which should probably be a `Set` by the way), will always be limited to 1 or 2 elements, so I think this is acceptable.